credentials/alts/utils.go

```diff
@@ -142,3 +144,20 @@ func AuthInfoFromPeer(p *peer.Peer) (AuthInfo, error) {
 	}
 	return altsAuthInfo, nil
 }
+
+// ClientAuthorizationCheck checks whether the client is authorized to access
+// the requested resources based on the given expected client service accounts.
+// This API should be used by gRPC server RPC handlers. This API should not be
+// used by clients.
+func ClientAuthorizationCheck(ctx context.Context, expectedServiceAccounts []string) error {
+	authInfo, err := AuthInfoFromContext(ctx)
+	if err != nil {
+		return status.Newf(codes.NotFound, "The context is not an ALTS-compatible context: %v", err).Err()
```

Comment:
If context is not compatible, should this also be a `PermissionDenied`?

Reply:
I can go either way. It is a permission deny caused by not found.

Reply:
To me, `NotFound` doesn't seem to be appropriate.

`NotFound` usually means the resource requested by the client is not found in the server.
But this is when necessary AuthInfo server needs is not found in the request.

Reply:
Change the code to `PermissionDenied`.